posthog/temporal/batch_exports/s3_batch_export.py

```diff
@@ -970,23 +982,456 @@ async def insert_into_s3_activity_from_stage(inputs: S3InsertInputs) -> RecordsC
+                                max_retry_delay, initial_retry_delay * (attempt**exponential_backoff_coefficient)
+                            )
+                            await self.logger.ainfo("Retrying part %s upload in %s seconds", part_number, retry_delay)
+                            await asyncio.sleep(retry_delay)
+                            continue
+                        else:
+                            raise
+
+                upload_time = time.time() - upload_start
+                part_info: CompletedPartTypeDef = {"ETag": response["ETag"], "PartNumber": part_number}
+
+                # Store completed part info
+                self.completed_parts[part_number] = part_info
+
+                # Calculate transfer speed
+                part_size_mb = len(data) / (1024 * 1024)
+                upload_speed_mbps = part_size_mb / upload_time if upload_time > 0 else 0
+
+                await self.logger.ainfo(
+                    "Finished uploading file number %s part %s with upload id %s. File size: %sMB, upload time: %s, speed: %s MB/s",
+                    self.current_file_index,
+                    part_number,
+                    self.upload_id,
+                    part_size_mb,
+                    upload_time,
+                    upload_speed_mbps,
+                )
+                return part_info
+
+        except Exception:
+            await self.logger.aexception(
+                "Failed to upload file number %s part %s with upload id %s",
+                self.current_file_index,
+                part_number,
+                self.upload_id,
+            )
+            raise
+        finally:
+            # Always release memory semaphore
+            self.memory_semaphore.release()
+            # Explicitly delete data to free memory immediately
+            del data
+
+    def _get_current_key(self) -> str:
+        """Generate the key for the current file"""
+        key_prefix = get_s3_key_prefix(self.s3_inputs)
+
+        try:
+            file_extension = FILE_FORMAT_EXTENSIONS[self.s3_inputs.file_format]
+        except KeyError:
+            raise UnsupportedFileFormatError(self.s3_inputs.file_format, "S3")
+
+        base_file_name = f"{self.s3_inputs.data_interval_start}-{self.s3_inputs.data_interval_end}"
+        # to maintain backwards compatibility with the old file naming scheme
+        if self.s3_inputs.max_file_size_mb is not None:
+            base_file_name = f"{base_file_name}-{self.current_file_index}"
+        if self.s3_inputs.compression is not None:
+            file_name = base_file_name + f".{file_extension}.{COMPRESSION_EXTENSIONS[self.s3_inputs.compression]}"
+        else:
+            file_name = base_file_name + f".{file_extension}"
+
+        key = posixpath.join(key_prefix, file_name)
+
+        if posixpath.isabs(key):
+            # Keys are relative to root dir, so this would add an extra "/"
+            key = posixpath.relpath(key, "/")
+
+        return key
+
```

Comment:
I think this the same as the `get_s3_key` function defined in line 143, why not use that?

Reply:
Unless I am missing something, it looks the same

Reply:
have switched to using the existing version